### PR TITLE
Persist DKG shares

### PR DESCRIPTION
**Pull request by makramkd**
This PR adds DKG share persistence support.

### CODE REVIEW

File: core/services/ocr2/plugins/dkg/persistence/db.go [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+	shareRecords []ocr2vrftypes.PersistentShareSetRecord,
+) error {
+	lggr := s.lggr.With(
+		"configDigest", hexutil.Encode(cfgDgst[:]),
+		"keyID", hexutil.Encode(keyID[:]))
+
+	start := time.Now()
+	defer func() {
+		lggr.Infow("Inserted DKG shares into DB", "duration", time.Since(start))
+	}()
```
Comment (jinhoonbang) [inline core/services/ocr2/plugins/dkg/persistence/db.go:54]: would it be helpful to have this as a prom metric also?

File: core/services/ocr2/plugins/dkg/persistence/db.go [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+			return fmt.Errorf("wrong number of bytes copied in hash (dealer:%s) %x: %d",
+				record.Dealer.String(), record.Hash[:], copied)
+		}
+
+		named = append(named, dkgShare{
+			ConfigDigest:         cfgDgst[:],
+			KeyID:                keyID[:],
+			Dealer:               record.Dealer.Marshal(),
+			MarshaledShareRecord: record.MarshaledShareRecord,
+			/* TODO/WTF: can't do "record.Hash[:]": this leads to store the last record's hash for all the records! */
```
Comment (jinhoonbang) [inline core/services/ocr2/plugins/dkg/persistence/db.go:84]: hmm that's very strange...

File: core/services/ocr2/plugins/dkg/persistence/db.go [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+	return &shareDB{
+		q:    pg.NewQ(db, lggr, cfg),
+		lggr: lggr,
+	}
+}
+
+// WriteShareRecords writes the provided (already encrypted)
+// share records to the Chainlink database.
+func (s *shareDB) WriteShareRecords(
+	ctx context.Context,
```
Comment (jinhoonbang) [inline core/services/ocr2/plugins/dkg/persistence/db.go:42]: seems like this isn't used. should we use it for the db write?

File: core/services/ocr2/plugins/dkg/persistence/db.go (hunk 1 above)
Comment (makramkd) [reply core/services/ocr2/plugins/dkg/persistence/db.go:54]: Yeah, good idea!

File: core/services/ocr2/plugins/dkg/persistence/db.go [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+	keyID [32]byte,
+	shareRecords []ocr2vrftypes.PersistentShareSetRecord,
+) error {
+	lggr := s.lggr.With(
+		"configDigest", hexutil.Encode(cfgDgst[:]),
+		"keyID", hexutil.Encode(keyID[:]))
+
+	start := time.Now()
+	defer func() {
+		lggr.Infow("Inserted DKG shares into DB", "duration", time.Since(start))
```
Comment (vreff) [inline core/services/ocr2/plugins/dkg/persistence/db.go:53]: Info or debug?

Comment (makramkd) [reply core/services/ocr2/plugins/dkg/persistence/db.go:53]: Should be debug, thanks for the catch.

File: core/store/migrate/migrations/0148_dkg_shares.sql [hunk 5]
```diff
@@ -0,0 +1,12 @@
+-- +goose Up
+CREATE TABLE dkg_shares(
+    config_digest bytea NOT NULL CHECK ( length(config_digest) = 32 ),
+    key_id bytea NOT NULL CHECK ( length(key_id) = 32 ),
+    dealer bytea NOT NULL CHECK ( length(dealer) = 1),
```
Comment (vreff) [inline core/store/migrate/migrations/0148_dkg_shares.sql:5]: hmm what if the committee is bigger (31 nodes)?

**Comment by vreff** [comment]
LGTM

File: core/services/ocr2/plugins/dkg/persistence/db.go (hunk 1 above)
Comment (makramkd) [reply core/services/ocr2/plugins/dkg/persistence/db.go]: Added.

**Review by vreff — DISMISSED** [review]

**Review by jinhoonbang — APPROVED** [review]
LGTM. Excited to get this bug fixed !

**Pull request merged**